src/cmd/services/m3coordinator/ingest/ingest.go

```diff
@@ -119,13 +124,15 @@ func (i *Ingester) Ingest(
 }
 
 type ingestOp struct {
-	s         storage.Appender
-	r         retry.Retrier
-	it        id.SortedTagIterator
-	p         pool.ObjectPool
-	m         ingestMetrics
-	attemptFn retry.Fn
-	ingestFn  func()
+	s             storage.Appender
+	r             retry.Retrier
+	it            id.SortedTagIterator
+	p             pool.ObjectPool
+	m             ingestMetrics
+	logger        log.Logger
+	logSampleRate float32
```

Comment:
You might want to use an atomic int so that the very first error always comes out?

i.e. 
```go
import "github.com/uber-go/atomic"

type thing struct {
  logEvery int32
  numLogs *atomic.Int32
}

func newThing(sampleRate float64) *thing {
	return &thing{numLogs: atomic.NewInt32(0), logEvery: int32(float64(1.0/sampleRate)}
}

func (t *thing) shouldLog() bool {
	return t.numLogs.Inc(1)-1%t.logEvery
}
```